src/scoped-typeorm/scoped-typeorm.module.ts

```diff
@@ -1,21 +1,29 @@
-import { DynamicModule, Module, Provider } from '@nestjs/common';
-import { REQUEST } from '@nestjs/core';
+import { DynamicModule, Module, Provider, Scope } from '@nestjs/common';
 import { getDataSourceToken, getRepositoryToken } from '@nestjs/typeorm';
 import { DataSource, QueryRunner } from 'typeorm';
-import { Request } from 'express';
-
-type ReqOrQueryRunner = { queryRunner: QueryRunner } | Request;
+import { transactionStorage } from './transaction-store';
+import { ScopedInterceptor } from './scoped.interceptor';
+import { APP_INTERCEPTOR } from '@nestjs/core';
 
 @Module({})
 export class ScopedTypeOrmModule {
   static forFeature(entities: any[]): DynamicModule {
     const providers = entities.map(
       (entity): Provider => ({
+        // We override the TypeORM repository for backward compatibility with the Nest TypeOrmModule
         provide: getRepositoryToken(entity),
-        inject: [REQUEST, getDataSourceToken()],
```

Comment:
Injecting the REQUEST automatically disqualifies this from being durable, because it has a dependency that's not durable.